src/bio/bio_xstream.c

```diff
@@ -1140,13 +1156,54 @@ init_blobstore_ctxt(struct bio_xs_context *ctxt, int tgt_id)
 		}
 	}
 
+	*ret_d_bdev = d_bdev;
+	*ret_dev_info = dev_info;
+
+	return 0;
+out:
+	D_ASSERT(dev_info != NULL);
+	smd_dev_free_info(dev_info);
+
+	return rc;
+}
+
+static int
+init_xs_blobstore_ctxt(struct bio_xs_context *ctxt, int tgt_id, enum smd_type st)
+{
+	struct bio_bdev		*d_bdev;
+	struct bio_blobstore	*bbs;
+	struct spdk_blob_store	*bs;
+	struct bio_xs_blobstore	*bxb;
+	struct smd_dev_info	*dev_info = NULL;
+	int			 rc;
+
+	D_ASSERT(!ctxt->bxc_ready);
+	D_ASSERT(ctxt->bxc_xs_blobstores[st] == NULL);
+
+	if (d_list_empty(&nvme_glb.bd_bdevs)) {
+		D_ERROR("No available SPDK bdevs, please check whether "
+			"VOS_BDEV_CLASS is set properly.\n");
+		return -DER_UNINIT;
+	}
+
+	ctxt->bxc_xs_blobstores[st] = alloc_xs_blobstore();
```

Comment:
How the owner xstream is decided now?

Reply:
This is same as before, owner xstream will be set in alloc_xs_blobstore